near-sdk/src/environment/env.rs

```diff
@@ -488,9 +488,23 @@ pub fn value_return(value: &[u8]) {
     unsafe { sys::value_return(value.len() as _, value.as_ptr() as _) }
 }
 /// Terminates the execution of the program with the UTF-8 encoded message.
+/// [`panic_str`] should be used as the bytes are required to be UTF-8
+#[deprecated(since = "4.0.0", note = "Use env::panic_str to panic with a message.")]
 pub fn panic(message: &[u8]) -> ! {
     unsafe { sys::panic_utf8(message.len() as _, message.as_ptr() as _) }
 }
+
+/// Terminates the execution of the program with the UTF-8 encoded message.
+pub fn panic_str(message: &str) -> ! {
+    unsafe { sys::panic_utf8(message.len() as _, message.as_ptr() as _) }
+}
+
+/// Aborts the current contract execution without a custom message.
+/// To include a message, use [`panic_str`].
```

Comment:
I believe you said this is to really minimize overhead. Should we put something in the comments for folks to know when they might want to use this? This might be wrong, but I was imagining something like, "This can be a useful approach for smart contracts that avoid using UTF-8 for optimization reasons, something something…"

Reply:
Has nothing to do with optimizing, as this has the same internal type as bytes, this just makes sure at compile time that the "bytes" are utf8 and is a more ergonomic type (easier to get to `&str` for logging messages and bytes gives no benefit since non-utf8 disallowed).